sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,252 @@
+   * query string values are logged.
+   */
+  additionalAllowedQueryParameters?: string[];
+}
+
+
+export enum TracingAttributesEnum {
+  Operation_Name="gen_ai.operation.name",
+  Request_Model="gen_ai.request.model",
+  System="gen_ai.system",
+  Error_Type="error.type",
+  Server_Port="server.port", // not use it for now
+  Request_Frequency_Penalty="gen_ai.request.frequency_penalty",
+  Request_Max_Tokens="gen_ai.request.max_tokens",
+  Request_Presence_Penalty="gen_ai.request.presence_penalty",
+  Request_Stop_Sequences="gen_ai.request.stop_sequences",
+  Request_Temperature="gen_ai.request.temperature",
+  Request_Top_K="gen_ai.request.top_k",
+  Request_Top_P="gen_ai.request.top_p",
+  Response_Finish_Reasons="gen_ai.response.finish_reasons",
+  Response_Id="gen_ai.response.id",
+  Response_Model="gen_ai.response.model",
+  Usage_Input_Tokens="gen_ai.usage.input_tokens",
+  Usage_Output_Tokens="gen_ai.usage.output_tokens",
+  Server_Address="server.address"
+}
+
+export type TracingAttributes = { [key in TracingAttributesEnum]?: string | number};
+
+/**
+ * A simple policy to create OpenTelemetry Spans for each request made by the pipeline
+ * that has SpanOptions with a parent.
+ * Requests made without a parent Span will not be recorded.
+ */
+export function tracingPolicy(): PipelinePolicy {
+
+  return {
+    name: tracingPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      if (!tracingClient) {
+        return next(request);
+      }
+
+      /* sample of request
+      request = {
+        url: "https://models.inference.ai.azure.com/chat/completions?api-version=2024-05-01-preview",
+        body: "{\"messages\":[{\"role\":\"system\",\"content\":\"You are a helpful assistant.\"},{\"role\":\"user\",\"content\":\"What is the capital of France?\"}],\"model\":\"Phi-3-medium-128k-instruct\",\"temperature\":1,\"max_tokens\":1000,\"top_p\":1}",
+        headers: {
+          _headersMap: {
+          },
+        },
+        method: "POST",
+        timeout: 0,
+        multipartBody: undefined,
+        formData: undefined,
+        disableKeepAlive: false,
+        proxySettings: undefined,
+        streamResponseStatusCodes: undefined,
+        withCredentials: false,
+        abortSignal: undefined,
+        tracingOptions: {
+        },
+        onUploadProgress: undefined,
+        onDownloadProgress: undefined,
+        requestId: "e89d3930-257d-4d96-a60b-216e5ecc8040",
+        allowInsecureConnection: false,
+        enableBrowserStreams: true,
+      }
+
+       */
+      if (!request.body)
+        return next(request);
+      const body = JSON.parse(request.body as string);
+
+      const spanAttributes: TracingAttributes = {
+        [TracingAttributesEnum.Operation_Name]: "operation name", ///TODD: don't know how to determine yet
+        [TracingAttributesEnum.Request_Model]: body.model,
```

Comment:
we also need to set "gen_ai.system", discussing what it should be here https://github.com/Azure/azure-sdk-for-python/pull/36890#discussion_r1728196605

Reply:
done